azure-kusto-ingest/src/queued_ingest.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use azure_core::base64;
+use azure_kusto_data::prelude::KustoClient;
+use rand::rngs::StdRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+
+use crate::client_options::QueuedIngestClientOptions;
+use crate::descriptors::BlobDescriptor;
+use crate::ingestion_blob_info::QueuedIngestionMessage;
+use crate::ingestion_properties::IngestionProperties;
+use crate::resource_manager::ResourceManager;
+
+/// Client for ingesting data into Kusto using the queued flavour of ingestion
+#[derive(Clone)]
+pub struct QueuedIngestClient {
+    resource_manager: Arc<ResourceManager>,
+}
+
+impl QueuedIngestClient {
+    /// Creates a new client from the given [KustoClient]
+    pub fn new(kusto_client: KustoClient) -> Self {
+        Self::new_with_client_options(kusto_client, QueuedIngestClientOptions::default())
+    }
+
+    /// Creates a new client from the given [KustoClient] and [QueuedIngestClientOptions]
+    /// This allows for customisation of the [ClientOptions] used for the storage clients
+    pub fn new_with_client_options(
+        kusto_client: KustoClient,
+        options: QueuedIngestClientOptions,
+    ) -> Self {
+        // TODO: add a validation check that the client provided is against the ingestion endpoint
```

Comment:
I don't think this is required - and if it is, it would likely involve changes in azure-kusto-data. 
For now I'll just document that the client provided should be against the ingestion URI